packages/app/src/components/swap/swap.tsx

```diff
@@ -57,20 +61,10 @@ const Swap: FC<Props> = ({
     .mul(11)
     .div(10);
 
-  const { state: _swapTx, send: swapExactAmountIn } = useContractFunction(
-    contracts[protocolName].swap,
-    "swapExactAmountIn"
-  );
+  const { account } = useEthers();
 
-  const { state: _approveTokenTx, send: approveUnderlyingTokens } = useContractFunction(
-    contracts[protocolName].token,
-    "approve"
-  );
-
-  const { state: _approveTenderTx, send: approveTenderTokens } = useContractFunction(
-    contracts[protocolName].tenderToken,
-    "approve"
-  );
+  const allowance = useTokenAllowance(tokenSendedAddress, account, addresses[protocolName].swap);
+  const isTokenApproved = allowance != null && constants.MaxUint256.div(2).lt(allowance);
```

Comment:
This doesn't check what we want. 

We want to check whether the user input is lesser or greater than the allowance. 


```suggestion
  const allowance: BigNumber | undefined = useTokenAllowance(tokenSendedAddress, account, addresses[protocolName].swap);
  const isTokenApproved = allowance?.gte(sendTokenAmount);
```

Reply:
just a minor thing, but optional chaining should only be used if we are okay with undefined as a result - I prefer to use booleans so I don't have to handle undefined in the typings.

Reply:
Unfortunately this is just how `useDapp` handles return types, I hear you. 

Later on we can perhaps wrap these hooks in our own custom hooks that does `return useTokenAllowance(...) || constants.Zero`

Reply:
I already started to use a custom hook for the next PR